rmb-sdk-go/peer/encoder/encoder.go

```diff
@@ -0,0 +1,30 @@
+package encoder
+
+import (
+	"fmt"
+)
+
+// Encoder interface for encoding data
+type Encoder interface {
+	Schema() string
+	Encode(interface{}) ([]byte, error)
+	Decode([]byte, interface{}) error
+}
+
+type Schema string
+
+// supported schemas
+const (
+	JSONSchema    = "application/json"
+	DefaultSchema = JSONSchema
+)
+
+// NewEncoder returns an encoder based on the given schema
+func NewEncoder(schema Schema) (Encoder, error) {
```

Comment:
Check the other comment regarding how the encoder need to be configured. 

In short, This means that user of this library cannot use custom encoding unless this library is modified (or forked) Instead the `WithEncoder` need to take an encoder object, not schema. Hence I (as a 3rd party) can implement my cutsom encoder type.